Mathlib/RingTheory/MvPowerSeries/Basic.lean

```diff
@@ -719,6 +720,81 @@ theorem coeff_prod [DecidableEq σ]
       simp only [add_right_inj] at huv
       exact h rfl huv.symm
 
+/-- The `d`th coefficient of a power of a multivariate power series
+is the sum, indexed by `finsuppAntidiag (Finset.range n) d`, of products of coefficients  -/
+theorem coeff_pow [DecidableEq σ] (f : MvPowerSeries σ R) {n : ℕ} (d : σ →₀ ℕ) :
+    coeff R d (f ^ n) =
+      ∑ l in finsuppAntidiag (Finset.range n) d,
+        ∏ i in Finset.range n, coeff R (l i) f := by
+  suffices f ^ n = (Finset.range n).prod fun _ ↦ f by
+    rw [this, coeff_prod]
+  rw [Finset.prod_const, card_range]
+
+/-- The degree of a monomial. -/
+def degree (d : σ →₀ ℕ) : ℕ := d.sum fun _ ↦ id
+
+@[simp]
+theorem degree_zero : degree (0 : σ →₀ ℕ) = 0 := by
+  simp only [degree, sum_zero_index]
+
+theorem degree_add (d d' : σ →₀ ℕ) :
+    degree (d + d') = degree d + degree d' := by
+  classical
+  simp only [degree, mem_union, mem_support_iff, ne_eq, id_eq, implies_true, sum_add_index]
+
+theorem degree_eq_zero_iff (d : σ →₀ ℕ) : degree d = 0 ↔ d = 0 := by
+  constructor <;>
+  intro hd
+  · ext x
+    simp only [Finsupp.coe_zero, Pi.zero_apply, ← Nat.lt_one_iff, ← not_le]
+    intro hx
+    apply Nat.not_add_one_le_zero 0
+    rw [zero_add, ← hd]
+    exact hx.trans
+      <| single_le_sum (fun _ _ ↦ zero_le _) (mem_support_iff.mpr <| Nat.not_eq_zero_of_lt hx)
+  · simp only [hd, degree_zero]
+
+/-- Vanishing of coefficients of powers of multivariate power series
+when the constant coefficient is nilpotent
+[N. Bourbaki, *Algebra {II}*, Chapter 4, §4, n°2, proposition 3][bourbaki1981] -/
+theorem coeff_eq_zero_of_constantCoeff_nilpotent [DecidableEq σ]
+    {f : MvPowerSeries σ R} {m : ℕ} (hf : constantCoeff σ R f ^ m = 0)
+    {d : σ →₀ ℕ} {n : ℕ} (hn : m + degree d ≤ n) :
+    coeff R d (f ^ n) = 0 := by
+  rw [coeff_pow]
+  apply sum_eq_zero
+  intro k hk
+  rw [mem_finsuppAntidiag] at hk
+  set s := (range n).filter fun i ↦ k i = 0 with hs_def
+  have hs : s ⊆ range n := filter_subset _ _
+  have hs' : ∀ i ∈ s, coeff R (k i) f = constantCoeff σ R f := by
+    intro i hi
```

Comment:
```suggestion
  have hs' (i : ℕ) (hi : i ∈ s) : coeff R (k i) f = constantCoeff σ R f := by
```

Reply:
I am pretty sure we don't. I keep it as a small project for students... but maybe it's time to do it.